java/maven/src/org/netbeans/modules/maven/queries/MavenReloadImplementation.java

```diff
@@ -0,0 +1,261 @@
+        if (a != null) {
+            toProcess.add(a);
+        }
+        toProcess.addAll(mp.getArtifacts());
+        
+        while ((a = toProcess.poll()) != null) {
+            if (!processed.add(a)) {
+                continue;
+            }
+            File pom = MavenFileOwnerQueryImpl.getInstance().getOwnerPOM(a.getGroupId(), a.getArtifactId(), a.getVersion());
+            if (pom != null) {
+                Project parentOwner = FileOwnerQuery.getOwner(FileUtil.toFileObject(pom));
+                // can't call getProjectState, since that may block 
+                FileObject parentPom = parentOwner.getProjectDirectory().getFileObject("pom.xml");
+                if (parentPom != null) {
+                    fileSet.add(parentPom);
+                }
+                NbMavenProject p2 = project.getLookup().lookup(NbMavenProject.class);
+                a = p2.getMavenProject().getParentArtifact();
+                if (a != null) {
+                    toProcess.add(a);
+                }
+                toProcess.addAll(p2.getMavenProject().getArtifacts());
+            }
+        }
+        return fileSet;
+    }
+    
+    private static class CF extends CompletableFuture<ProjectStateData<MavenProject>> {}
+
+    static ProjectReload.Quality getProjectQuality(MavenProject mp) {
+        if (MavenProjectCache.isFallbackproject(mp)) {
+            // could not load at all.
+            MavenProject fallback = MavenProjectCache.getPartialProject(mp);
+            return fallback != null ? 
+                    ProjectReload.Quality.BROKEN:
+                    ProjectReload.Quality.NONE;
+        }
+        if (!MavenProjectCache.getFakedArtifacts(mp).isEmpty()) {
+            return ProjectReload.Quality.LOADED;
+        }
+        return ProjectReload.Quality.RESOLVED;
+    }
+    
+    private ProjectStateData createStateData(MavenProject mp) {
+        ProjectStateData d;
+        
+        ProjectStateBuilder builder = ProjectStateData.builder(getProjectQuality(mp)).
+                files(findProjectFiles()).
+                timestamp(MavenProjectCache.getLoadTimestamp(mp));
+        builder.data(mp);
+        builder.attachLookup(Lookups.fixed(mp));
+        d = builder.build();
+        synchronized (this) {
+            ProjectStateData d2 = lastData.get();
+            if (d2 != null) {
+                if (d2.getProjectData() == mp) {
+                    return d2;
+                }
+                d2.fireChanged(true, false);
+            }
+            lastData = new WeakReference<>(d);
+        }
+        return d;
+    }
+
+    @NbBundle.Messages({
+        "# {0} - project name",
+        "ERR_UnprimedInOfflineMode=Priming build for {0} is required, but offline operation was requested.",
+        "# {0} - project name",
+        "# {1} - parent gav",
+        "ERR_ParentPomMissing=Project {0} is missing its parent artifact ({1}) is missing and offline operation was requesed"
```

Comment:
Typo - "requesed" -> "requested".

There are a few others in the docs rather than messages.

Reply:
fixed in 4c091822130a